pkg/dockerregistry/server/auth.go

```diff
@@ -21,6 +21,19 @@ import (
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
+type deferredErrors map[string]error
+
+func (d deferredErrors) Add(namespace string, name string, err error) {
+	d[namespace+"/"+name] = err
```

Comment:
I'd rather define `(*repository).Name()` returning `string` and go without helpers.